tests/terraform/createcluster_test.go

```diff
@@ -0,0 +1,231 @@
+			defer removeWorkload("ingress.yaml", kubeConfigFile)
+
+			Eventually(func(g Gomega) {
+				cmd := "kubectl get pods -n " + namespace + " -o=name -l k8s-app=nginx-app-ingress --field-selector=status.phase=Running --kubeconfig=" + kubeConfigFile
+				res, err := runCommand(cmd)
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(res).Should(ContainSubstring("test-ingress"))
+			}, "240s", "5s").Should(Succeed())
+
+			var ingressIps []string
+			nodes, err := workerNodes(kubeConfigFile, false)
+			if err != nil {
+				fmt.Println("Error retrieving nodes: ", err)
+			}
+			Eventually(func(g Gomega) {
+				ingressIps, err = fetchIngressIP(namespace, kubeConfigFile)
+				g.Expect(err).NotTo(HaveOccurred(), "Ingress ip is not returned")
+				g.Expect(len(ingressIps)).To(Equal(len(nodes)), "Number of ingress IPs should match the number of nodes")
+			}, "240s", "5s").Should(Succeed())
+
+			for _, ip := range ingressIps {
+				cmd := "curl -s --header host:foo1.bar.com" + " http://" + ip + "/name.html"
+				Eventually(func(g Gomega) {
+					res, err := runCommand(cmd)
+					g.Expect(err).NotTo(HaveOccurred())
+					g.Expect(res).Should(ContainSubstring("test-ingress"))
+				}, "240s", "5s").Should(Succeed())
+			}
+		})
+
+		It("Verifies Daemonset", func() {
+			_, err := deployWorkload("daemonset.yaml", kubeConfigFile)
+			Expect(err).NotTo(HaveOccurred(), "Daemonset manifest not deployed")
+			defer removeWorkload("daemonset.yaml", kubeConfigFile)
+
+			nodes, _ := workerNodes(kubeConfigFile, false)
+			pods, _ := pods(kubeConfigFile, false)
+
+			Eventually(func(g Gomega) {
+				count := countOfStringInSlice("test-daemonset", pods)
+				g.Expect(count).Should((Equal(len(nodes))), "Daemonset pod count does not match node count")
+			}, "420s", "10s").Should(Succeed())
+		})
+
+		It("Verifies dns access", func() {
+			namespace := "auto-dns"
+			_, err := deployWorkload("dnsutils.yaml", kubeConfigFile)
+			Expect(err).NotTo(HaveOccurred(), "dnsutils manifest not deployed")
+			defer removeWorkload("dnsutils.yaml", kubeConfigFile)
+
+			Eventually(func(g Gomega) {
+				cmd := "kubectl get pods dnsutils " + "-n " + namespace + " --kubeconfig=" + kubeConfigFile
+				res, _ := runCommand(cmd)
+				g.Expect(res).Should(ContainSubstring("dnsutils"))
+				g.Expect(res).Should(ContainSubstring("Running"))
+			}, "420s", "2s").Should(Succeed())
+
+			Eventually(func(g Gomega) {
+				cmd := "kubectl -n " + namespace + " --kubeconfig=" + kubeConfigFile + " exec -t dnsutils -- nslookup kubernetes.default"
+				res, _ := runCommand(cmd)
+				g.Expect(res).Should(ContainSubstring("kubernetes.default.svc.cluster.local"))
+			}, "420s", "2s").Should(Succeed())
+		})
+	})
+})
+
+var _ = AfterEach(func() {
+	failed = failed || CurrentSpecReport().Failed()
+})
+
+var _ = BeforeEach(func() {
+	failed = failed || CurrentSpecReport().Failed()
+	if *destroy {
+		Skip("Cluster is being Deleted")
+	}
+})
```

Comment:
What is this section trying to achieve? Are you simply trying to check for errors and then skip to the end if one is found?

Reply:
This is a good callout...
To be honest, the `AfterEach` can probably be removed entirely, and the update to the `failed` variable in the `BeforeEach` can be also, since they're just there for this pretty irrelevant block at the end of the suite:
```
	if failed {
		fmt.Println("FAILED!")
	} else {
		fmt.Println("PASSED!")
	}
```
The `destroy` block in the `BeforeEach` is there primarily for local runs of this. It purposely skips all the specs and just destroys all the resources created by terraform at the end of the suite:
```
	if *destroy {
		status, err := buildCluster(&testing.T{}, *tfVars, *destroy)
		Expect(err).NotTo(HaveOccurred())
		Expect(status).To(Equal("cluster destroyed"))
	}
```

TLDR: I'll remove the unnecessary bits 👍